corehq/apps/hqwebapp/static/hqwebapp/js/components/pagination.js

```diff
@@ -81,6 +85,7 @@ hqDefine('hqwebapp/js/components/pagination', [
                 self.goToPage(Math.max(self.currentPage() - 1, 1), e);
             };
             self.goToPage = function (page, e) {
+                if (!self.isReady()) return;
```

Comment:
I'm not sure this lives here. Why not put it in the `goToPage` function of the exports code?

Reply:
That is an excellent idea!